### PR TITLE
LoRaWAN Registration (TTS & Helium)

### PR DESCRIPTION
**Pull request by gijsio**

### CODE REVIEW

File: content/gettingstarted/registration/lora/ttn.md [hunk 1]
```diff
@@ -1,85 +1,82 @@
...
-In order to register your device to connect to the things network, you must first create an application for these devices to belong to. This way the Network will know where to send the devices data to.
+![](/gitbook/assets/lorawan/tts/index.png)
 
-Selecting the `Applications` tab at the top of the TTN console, will bring up a screen for registering applications. Click register and a new page, similar to the one below, will open.
+![](/gitbook/assets/lorawan/tts/account.png)
 
-![](/gitbook/assets/ttn-5.png)
+Once an account has been registered, you can [create an application](#register-a-node) for your nodes, or [register a gateway](#register-a-gateway).
+
+## Registera node
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:26]: missing space

File: content/gettingstarted/registration/lora/ttn.md [hunk 2]
```diff
@@ -1,85 +1,82 @@
...
+
+In order to register your device, you must first create an application for these devices to belong to. This way the Network will know where to send the devices data to.
+
+Selecting the `Applications` tab at the top of the TTN console, will bring up a screen for registering applications. 
+
+![](/gitbook/assets/lorawan/tts/application.png)
 
 Enter a unique `Application ID` as well as a Description & Handler Registration.
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:34]: ?Handler Registration?

File: content/gettingstarted/registration/lora/ttn.md [hunk 3]
```diff
@@ -1,85 +1,82 @@
...
 
 Now the Pycom module nodes can be registered to send data up to the new Application.
 
-## Register a Device
-
-To connect nodes to a things network gateway, devices need to be added to the application. To do this, navigate to the `Devices` tab on the `Application` home page and click the `Register Device` button.
+You'll need to register your devices as nodes to the application. Click the button `+ Add device`. Next, you'll need to enter the specifics. You can choose either `OTAA` or `ABP` as activation methods. Learn more about the difference [here](https://www.thethingsindustries.com/docs/devices/abp-vs-otaa/). In this application we'll choose `OTAA`. Select `LoRaMAC V1.0.2` and check whether the region servers are set correctly.
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:38]: You'll need to register your devices as nodes to the application -> Now you can add your devices.

File: content/gettingstarted/registration/lora/ttn.md [hunk 4]
```diff
@@ -1,85 +1,82 @@
...
 
-You can also setup your Pycom module to act as a gateway with The Things Network. The code required to do this can be found [here](/tutorials/networks/lora/lorawan-nano-gateway).
+Great! Now your device is registered with TTS. You can now use the credentials in the [LoRaWAN OTAA](/tutorials/networks/lora/lorawan-otaa/) example.
 
-Inside the TTN Console, there are two options, `Applications` and `Gateways`. Select `Gateways` and then click on `register Gateway`. This will allow for the set up and registration of a new nano-gateway.
+## Register a Gateway
 
-![](/gitbook/assets/ttn-2.png)
+A LoRaWAN gateway receives the LoRa packets sent out by your (and other people's) nodes and forwards them to the cloud. Next to that, the gateway is able to provide a downlink to your nodes as well. In this case, we will forward the packets to TTS. From there, you can view the traffic coming through your gateway, and view the received messages in your nodes' application. You can also setup your LoRa-enabled Pycom module or Pygate to act as a LoRaWAN gateway. Below we will setup the Gateway in TTS. Have a look at the following pages for more information concerning the device setup:
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:61]: I don't understand the "In this case, we will ... " sentence. This whole section is about registering a gateway, right?!

File: content/gettingstarted/registration/lora/ttn.md [hunk 5]
```diff
@@ -1,85 +1,82 @@
...
-
-Most LoRaWAN network servers expect a Gateway ID in the form of a unique 64-bit hexadecimal number (called a EUI-64). The recommended practice is to produce this ID from your board by expanding the WiFi MAC address (a 48-bit number, called MAC-48). You can obtain that by running this code prior to configuration:
+Here, it is important we add the `Gateway ID` and `Gateway EUI`. The first can be an identifying string of characters unique to the gateway. The latter is what we'll use in the gateway configuration on the device, and exists of 16 hexadecimal characters. You can use the following to generate the `Gateway EUI`:
 
 ```python
-
  from network import WLAN
  import binascii
  wl = WLAN()
  binascii.hexlify(wl.mac())[:6] + 'FFFE' + binascii.hexlify(wl.mac())[6:]
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:75]: Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
TypeError: object with buffer protocol required

File: content/gettingstarted/registration/lora/helium.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+To add a device, click `Devices` in the left hand menu bar and `Add device` on the following page. The following screen will pop up:
+
+![](/gitbook/assets/lorawan/helium/device.png)
+
+Give your device a human-readable name. All other credentials should be filled in already. After successful creation, the device will be shown. You can use the credentials in the [LoRaWAN OTAA](/tutorials/networks/lora/lorawan-otaa/) example.
+
+## Register a gateway
+
+For more information about registering a gateway, we'd like to forward you to the [Helium Docs](https://docs.helium.com/use-the-network/build-a-packet-forwarder#packet-forwarder-architecture). Note that the Pygate uses a Semtech UDP forwarding architecture, and you will need to configure your own miner.
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/helium.md:29]: Note that the Pygate uses a Semtech UDP forwarding architecture, and you will need to configure your own miner.

I don't get it

**Review by peter-pycom — APPROVED** [review]
@gijsio  please note my review comments. I'll merge as is, but please follow up.

**Comment by peter-pycom** [comment]
Also the general register LoRaWAN page needs an overhaul gettingstarted/registration/lora/, e.g. refers to mac/dev_eui but plucks it out of the air

File: content/gettingstarted/registration/lora/ttn.md [hunk 7]
```diff
@@ -1,85 +1,82 @@
...
 
-In order to use The Things Network (TTN) you should navigate to their website and create/register an account. Enter a username and an email address to verify with their platform.
+> TheThingsNetwork (TTN) is migrating to TheThingsStack Community edition in 2021. Any applications running on TTN should be migrated to the new environment.
 
-![](/gitbook/assets/ttn-1.png)
+The Things Stack (TTS) is a LoRaWAN service that allows you to send and receive LoRa packets from your nodes. Registering your LoRa-enabled Pycom nodes and gateways with TTS gives access to their IoT infrastructure and integrate your devices with the cloud. For more information about The Things Stack and LoRaWAN in general, you can visit their documentation at [https://www.thethingsindustries.com/docs/](https://www.thethingsindustries.com/docs/). On this page, we will cover:
 
-Once an account has been registered, you can register your Pycom module as either a node or a nano-gateway. The steps below will detail how to do this.
+* [Create an account](#create-an-account)
+* [Register a node](#register-a-node)
```
Comment (peter-pycom) [inline content/gettingstarted/registration/lora/ttn.md:14]: link doesn't work. how are we doing with link-checker?

File: content/gettingstarted/registration/lora/ttn.md (hunk 4 above)
Comment (gijsio) [reply content/gettingstarted/registration/lora/ttn.md:61]: In my head, the emphasis was more on the 'in THIS case, we will forward packets to TTS', but it does not make as much sense anymore

File: content/gettingstarted/registration/lora/ttn.md (hunk 5 above)
Comment (gijsio) [reply content/gettingstarted/registration/lora/ttn.md:75]: Yes, fixed!

**Pull request merged**